### PR TITLE
Package salt master container

### PR DESCRIPTION
**Pull request by sayf-eddine-scality**
Salt-master image is locally built and shipped with the rest of the
docker images

Fixes: #606

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -281,6 +282,13 @@ $(ISO_ROOT)/images/$(NGINX_IMAGE_NAME)-$(NGINX_IMAGE_VERSION).tar.gz: \
 	IMAGE=$(NGINX_IMAGE)
 $(ISO_ROOT)/images/$(NGINX_IMAGE_NAME)-$(NGINX_IMAGE_VERSION).tar.gz: \
 	IMAGE_TAG=$(NGINX_IMAGE_NAME):$(NGINX_IMAGE_VERSION)
+$(ISO_ROOT)/images/$(SALT_MASTER_IMAGE_NAME)-$(SALT_MASTER_IMAGE_VERSION).tar.gz:
+	mkdir -p $(dir $@)
+	docker build -t "$(SALT_MASTER_IMAGE_NAME):$(SALT_MASTER_IMAGE_VERSION)" images/salt-master
+	$(eval IMAGE=$(shell docker image inspect $(SALT_MASTER_IMAGE_NAME) --format {{.Id}}) )
+	$(eval IMAGE_TAG=$(SALT_MASTER_IMAGE_NAME):$(SALT_MASTER_IMAGE_VERSION) )
+	docker tag $(IMAGE) $(IMAGE_TAG)
```
Comment (thomasdanan) [inline Makefile]: Is there a need to do it in 2 steps ? setting the tag during docker build is not enough ?

Comment (sayf-eddine-scality) [reply Makefile]: I started doing it on two steps like the provided images but yes makes more sense to jump the tagging phase

**Review by thomasdanan — APPROVED** [review]

**Comment by sayf-eddine-scality** [comment]
@bert-e approve

File: Makefile [hunk 2]
```diff
@@ -68,7 +68,8 @@ ALL = \
 	$(ISO_ROOT)/images/$(KUBE_CONTROLLER_MANAGER_IMAGE_NAME)-$(KUBE_CONTROLLER_MANAGER_IMAGE_VERSION).tar.gz \
 	$(ISO_ROOT)/images/$(KUBE_PROXY_IMAGE_NAME)-$(KUBE_PROXY_IMAGE_VERSION).tar.gz \
 	$(ISO_ROOT)/images/$(KUBE_SCHEDULER_IMAGE_NAME)-$(KUBE_SCHEDULER_IMAGE_VERSION).tar.gz \
-	$(ISO_ROOT)/images/$(NGINX_IMAGE_NAME)-$(NGINX_IMAGE_VERSION).tar.gz
+	$(ISO_ROOT)/images/$(NGINX_IMAGE_NAME)-$(NGINX_IMAGE_VERSION).tar.gz \
+	$(ISO_ROOT)/images/$(SALT_MASTER_IMAGE_NAME)-$(SALT_MASTER_IMAGE_VERSION).tar.gz
```
Comment (NicolasT) [inline Makefile:72]: In future, please always keep a trailing `\` on the last line, and an empty line below (as there is one right now).

Rationale: when adding a new value to this list, one wouldn't need to edit two lines (like above), but only add one, which is easier to validate/review.

File: Makefile [hunk 3]
```diff
@@ -281,6 +282,11 @@ $(ISO_ROOT)/images/$(NGINX_IMAGE_NAME)-$(NGINX_IMAGE_VERSION).tar.gz: \
 	IMAGE=$(NGINX_IMAGE)
 $(ISO_ROOT)/images/$(NGINX_IMAGE_NAME)-$(NGINX_IMAGE_VERSION).tar.gz: \
 	IMAGE_TAG=$(NGINX_IMAGE_NAME):$(NGINX_IMAGE_VERSION)
+$(ISO_ROOT)/images/$(SALT_MASTER_IMAGE_NAME)-$(SALT_MASTER_IMAGE_VERSION).tar.gz:
+	mkdir -p $(dir $@)
+	$(eval IMAGE_TAG=$(SALT_MASTER_IMAGE_NAME):$(SALT_MASTER_IMAGE_VERSION) )
```
Comment (NicolasT) [inline Makefile:287]: There's no real reason to `eval` here. Just use the relevant values in-place below.

File: container_images.mk [hunk 4]
```diff
@@ -25,3 +25,8 @@ KUBE_SCHEDULER_IMAGE_VERSION = 1.11.7
 NGINX_IMAGE = nginx@sha256:dd2d0ac3fff2f007d99e033b64854be0941e19a2ad51f174d9240dda20d9f534
 NGINX_IMAGE_NAME = nginx
 NGINX_IMAGE_VERSION = 1.15.8
+
+# Custom built images
+SALT_MASTER_IMAGE = ""
+SALT_MASTER_IMAGE_NAME = salt-master
+SALT_MASTER_IMAGE_VERSION = 1.0.0
```
Comment (NicolasT) [inline container_images.mk:32]: No. This depends on the version of `salt-master` we build. This can be passed in as an `ARG` during the image build phase.

**Pull request merged**